llvm/lib/BinaryFormat/ELF.cpp

```diff
@@ -200,6 +200,71 @@ uint16_t ELF::convertArchNameToEMachine(StringRef Arch) {
       .Default(EM_NONE);
 }
 
+uint16_t ELF::convertTripleArchTypeToEMachine(Triple::ArchType ArchType) {
+  switch (ArchType) {
+  case Triple::UnknownArch:
+  default:
+    return EM_NONE;
```

Comment:
Can we leave `default` out? I think that we want a covered switch?

Reply:
I don't know that every ArchType has an e_machine mapping. If they do, I don't know what they all are.
I covered all the ones where I could figure out the mapping quickly myself.  If you know more mappings then we can add them here now.  But handling most rather than very few is strictly an improvement over the status quo, so I would like not to have it held up because most is not all.